pkg/challenges/challenges.go

```diff
@@ -40,11 +40,19 @@ const (
 	KubernetesValue
 )
 
+type WorkflowResult struct {
+	// Additional information associated with a Github Workflow
+	Trigger string
+	Sha     string
+}
+
 type ChallengeResult struct {
 	Issuer    string
 	TypeVal   ChallengeType
 	PublicKey crypto.PublicKey
 	Value     string
+	// Extra information, non-empty for a Github workflow
+	WorkflowInfo WorkflowResult
```

Comment:
I don't love that we're plumbing more and more stuff through these, but that's clearly the status quo, so I don't want to block this for it.

I wonder if we could plumb it through here as our own "extensions" type, and have the specific CA implementations translate all the extensions that way 🤔

Reply:
I may or may not be understanding this, but I think I agree -- we already have type information in the struct, and really this field should be something like Extensions(ForSpecificTypeVal). Some kind of interface that returns a string that can be serialized into an OID extension. Is that what you meant?

Reply:
PTAL! I decided a map would probably be better -- the difference CAs check for type and use the info in the map to create extensions.